buy_together_app/models.py

```diff
@@ -1,3 +1,50 @@
-# from django.db import models
+from django.db import models
+from django.core.validators import MinLengthValidator
+from django.core.exceptions import ValidationError
 
-# Create your models here.
+
+class Product(models.Model):
+    qr_code = models.CharField(max_length=32, validators=[MinLengthValidator(32)], primary_key=True)
+    product_name = models.CharField(max_length=32, validators=[MinLengthValidator(1)])
+    description = models.TextField(validators=[MinLengthValidator(1)])
+
+    @staticmethod
+    def filter_qr(qr):
+        if qr:
+            return Product.objects.filter(qr_code=qr)
+        return Product.objects.all()
+
+    @staticmethod
+    def filter_name(nm):
+        if nm:
+            return Product.objects.filter(product_name=nm)
+        else:
+            return []
+
+    @staticmethod
+    def filter_description(desc):
+        if desc:
+            return Product.objects.filter(description=desc)
+        else:
+            return []
+
+    @staticmethod
+    def get_all_products():
+        return Product.objects.all()
+
+    @staticmethod
+    def save_product(prd):
+        prd.save()
```

Comment:
I don't think this (static) method should get an already created object, otherwise it doesn't have to be static and we can just use `self`. it should be a kind-of constructor for the object, aka - performing validations (if such are needed), then create the object save it and return it.

and can you please explain the `full_clean` method for me?

Reply:
1 Done
2
Validating objects
There are four steps involved in validating a model:
Validate the model as a whole - [Model.clean()]
Validate the field uniqueness - [Model.validate_unique()]
Validate the constraints - [Model.validate_constraints()]
All four steps are performed when you call a model’s full_clean() method.

Done